infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/TopologyDescriptor.java

```diff
@@ -48,81 +70,78 @@ public class TopologyDescriptor {
     @Getter
     private NodeDescriptor localNodeDescriptor;
 
-    /**
-     * Constructor
-     *
-     * @param topologyMessage   topology message
-     * @param localNodeId       the identifier of this node
-     */
-    public TopologyDescriptor(TopologyConfigurationMsg topologyMessage, String localNodeId) {
-        this.topologyConfigId = topologyMessage.getTopologyConfigID();
-        this.sinkClusters = new HashMap<>();
-        this.sourceClusters = new HashMap<>();
-        this.invalidClusters = new HashMap<>();
-
-        for (ClusterConfigurationMsg clusterConfig : topologyMessage.getClustersList()) {
-            ClusterDescriptor cluster = new ClusterDescriptor(clusterConfig);
-            if (clusterConfig.getRole() == ClusterRole.SOURCE) {
-                sourceClusters.put(cluster.getClusterId(), cluster);
-            } else if (clusterConfig.getRole() == ClusterRole.SINK) {
-                sinkClusters.put(cluster.getClusterId(), cluster);
-            } else {
-                invalidClusters.put(cluster.getClusterId(), cluster);
-            }
-        }
-
-        setLocalDescriptor(localNodeId);
-    }
 
     /**
      * Constructor
      *
-     * @param topologyConfigId topology configuration identifier (epoch)
-     * @param sourceClusters source cluster's
-     * @param sinkClusters sink cluster's
+     * @param topologyConfigId  topology ID
+     * @param localNodeId      the identifier of this node
+     * @param remoteSinkToReplicationModel  remote Sink to local cluster and corresponding replication models
+     * @param remoteSourceToReplicationModel  remote Source to local cluster and corresponding replication models
+     * @param remoteClusterEndpoints remote clusters to which the local cluster will initiate connection
+     * @param allClusters     all remote clusters present in the topology
      */
-    public TopologyDescriptor(long topologyConfigId, @NonNull List<ClusterDescriptor> sourceClusters,
-                              @NonNull List<ClusterDescriptor> sinkClusters, String localNodeId) {
+    public TopologyDescriptor(long topologyConfigId, String localNodeId,
+                              Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSinkToReplicationModel,
+                              Map<ClusterDescriptor, Set<LogReplication.ReplicationModel>> remoteSourceToReplicationModel,
+                              Set<ClusterDescriptor> remoteClusterEndpoints, Set<ClusterDescriptor> allClusters) {
```

Comment:
no, thats not true most of the times with the new usecases.
`allClusters` contains all the clusters in the topology. If the current node is the connection initiator, it can be required to connection to just 1 remote even if there are 5 remotes in the topology, and also can be connection endpoint for other nodes.